cpp/tests/io/json_tree.cpp

```diff
@@ -361,9 +421,121 @@ tree_meta_t2 get_tree_representation_cpu(device_span<PdaTokenT const> tokens_gpu
           std::move(node_range_end)};
 }
 
+std::tuple<std::vector<NodeIndexT>, std::vector<size_type>> records_orient_tree_traversal_cpu(
+  host_span<SymbolT const> input, tree_meta_t2 const& tree, rmm::cuda_stream_view stream)
+{
+  // // move tree representation to cpu
+  // tree_meta_t2 tree = to_cpu_tree(d_tree, stream);
+  std::vector<NodeIndexT> node_ids(tree.parent_node_ids.size());
+  std::iota(node_ids.begin(), node_ids.end(), 0);
+
+  if (std::getenv("CUDA_DBG_DUMP") != nullptr) {
+    for (int i = 0; i < int(tree.node_range_begin.size()); i++) {
+      printf("%3s ",
+             std::string(input.data() + tree.node_range_begin[i],
+                         tree.node_range_end[i] - tree.node_range_begin[i])
+               .c_str());
+    }
+    printf(" (JSON)\n");
+    print_vec(tree.node_categories, "node_categories", to_cat);
+    print_vec(node_ids, "cpu.node_ids", to_int);
+  }
+
+  // print_vec(tree.parent_node_ids, "tree.parent_node_ids (before)");
+  constexpr NodeIndexT top_node = -1;  // TODO FIXME: uint32_t so, -1 is not legal.
```

Comment:
how about

```suggestion
  constexpr NodeIndexT top_node = ~uint32_t{};
```

Reply:
I think the comment is obsolete. We switched `NodeIndexT` to alias `size_type`, which is `int32_t`. 
```suggestion
  constexpr NodeIndexT top_node = -1; 
```